pkg/reconciler/events/storage/storage.go

```diff
@@ -193,12 +202,24 @@ func (r *Reconciler) deleteNotification(ctx context.Context, storage *v1alpha1.C
 	// Load the Bucket.
 	bucket := client.Bucket(storage.Spec.Bucket)
 
+	//Check whether Bucket exists or not
+	if _, err := bucket.Attrs(ctx); err != nil {
+		if err == ErrBucketNotExist {
+			logging.FromContext(ctx).Desugar().Info("Bucket doesn't exist", zap.String("bucketName", storage.Spec.Bucket), zap.Error(err))
+			return nil
```

Comment:
nit: Just add a comment saying that if the bucket was already deleted, then we should proceed...